### ISSUE TITLE
[Issue] Error message of missing tagging, when infrastructure is provided via Terraform

### ISSUE THREAD

**Issue by lechnerc77** [Azure/azure-dev#1439]
- [X] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**

```bash
azd version 0.5.0-beta.3 (commit 1c23678aade587d5b9685a256a3c1d2d7f4b42e0)
```

**Output from `az version`** 
Run `az version` and copy and paste the output here (minimum required version is 2.38.0): 

```json
{
  "azure-cli": "2.39.0",
  "azure-cli-core": "2.39.0",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {}
}
```

**Describe the bug**

When deploying an app and the tagging of the resource does not match the information of the `azure.yaml` file an error message is displayed. 

```bash
ERROR: creating project: creating service sap-cloud-sdk-api: resource not found: unable to find a resource tagged with 'azd-service-name: sap-cloud-sdk-api'. Ensure the service resource is correctly tagged in your bicep files, and rerun provision
```

This error message is very helpful, but it focuses on `bicep` as infrastructure provisioning mechanism. This should be stated either in a more generic way or it should take into account the infrastructure provisioning type defined in `azure.yaml`. This can cause confusions using Terraform. 

**To Reproduce**

- Configure the setup of an app via Terraform
- Change the service name in the `azure.yaml`, so that it does not match the tagging in the Terraform file
- Execute `azd up`

The deployment after the provisioning of the infrastructure will fail stating that:

**Expected behavior**

The error message should either take into account the infrastructure provisioning type (Bicep or Terraform) or be more generic like:

```bash
Ensure the service resource is correctly tagged in your bicep/Terraform files, and rerun provision
```

**Environment**

n/a

**Additional context**

- Original issue that introduced the additional check for tagging <https://github.com/Azure/azure-dev/issues/1037>
- PR containing the fix <https://github.com/Azure/azure-dev/pull/1181>
- Screenshot of error message when deploying via Terraform:

![Screenshot 2023-01-22 160928](https://user-images.githubusercontent.com/22294087/213923722-d645fd1f-27f9-4efd-aefb-7e4d58c81165.png)

**Comment by rajeshkamal5050** [comment]
@wbreza can we add a fix to have the error message specific in the context the IaC in execution?

**Comment by weikanglim** [comment]
Fixed with #1822
